frontend/src/components/object/DeleteObjectButton.vue

```diff
@@ -0,0 +1,50 @@
+<script setup lang="ts">
+// Types
+import { ButtonMode } from '@/interfaces/common/enums';
+import type { PropType } from 'vue';
+
+// PrimeVue / Fonts
+import Button from 'primevue/button';
+import { useConfirm } from 'primevue/useconfirm';
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
+
+const confirm = useConfirm();
+
+// Props
+const props = defineProps({
+  mode: {
+    type: String as PropType<ButtonMode>,
+    required: true,
+  },
+  ids: {
+    type: Array<string>,
+    required: true,
+  },
+});
+
+// Deletion
+const confirmDelete = () => {
+  confirm.require({
+    message: `Please confirm that you want to delete the selected ${props.ids.length > 1 ? 'files' : 'file'}`,
+    header: `Delete ${props.ids.length > 1 ? 'items' : 'item'}`,
+    acceptLabel: 'Confirm',
+    rejectLabel: 'Cancel',
+    accept: () => {
+      // TODO: Delete items
+      alert(`Not Implemented Yet (IDs ${props.ids})`);
```

Comment:
i guess you might have to update the store to remove the objects after their deleted.
by the way, by default calling the COMS DELETE /object/:objectId does a 'soft delete'..
i dont think we need to let users recover soft-deletes through the UI.. Maybe for now we only do hard-deletes which would be done by calling DELETE /object/:objectId?versionId=<latest version>
there's more info here: https://docs.aws.amazon.com/AmazonS3/latest/userguide/DeletingObjectVersions.html

Reply:
Yeah delete will likely wait until all are done and then re-fetch the table. Implementing next and will see what to do.

Reply:
We've got a whole host of soft-delete/hard-delete considerations we'll have to do at some point as well down the line. For now I think it's sufficient to just soft-delete and call it a day, as that's what's generally happening on S3Browser when you delete a file there anyways if I'm not mistaken.